install/helm/gloo/templates/7-gateway-proxy-deployment.yaml

```
@@ -70,10 +70,11 @@ spec:
         sidecar.istio.io/inject: "true"
           {{- end }}
         {{- end }}
-{{ $annotationExist := false}}
-{{- if or ($spec.podTemplate.extraAnnotations) (.Values.global.glooStats.setDatadogAnnotations) }}
-{{ $annotationExist = true}}
       annotations:
+        # This annotation was introduced to resolve https://github.com/solo-io/gloo/issues/8392
+        # It triggers a new rollout of the gateway proxy if the values in this config map changes.
+        checksum/{{ $name | kebabcase }}-envoy-config: {{ $gatewaySpec.checksum }}
+{{- if or ($spec.podTemplate.extraAnnotations) (.Values.global.glooStats.setDatadogAnnotations) }}
```

```suggestion
        {{- if or ($spec.podTemplate.extraAnnotations) (.Values.global.glooStats.setDatadogAnnotations) }}
```
the indent on the corresponding `end` should match as well